### PR TITLE
Set sensitive values as secret - Part 1

### PR DESCRIPTION
**Pull request by taylor-swanson**
## Proposed commit message

- Set sensitive values as secret in integrations.
- Updated package-spec to 3.0.2 in most packages to leverage secrets validation checks.
- Certain packages required fixes to mappings (due to missing or incorrect mappings).
- Certain packages were not able to go to package-spec 3.0.2 due issues with dynamic mappings. These have been held back at 3.0.0.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates elastic/security-team#7388

### CODE REVIEW

**Comment by mrodm** [comment]
Hi @taylor-swanson, please update your branch with the latest contents from main branch. There was an important PR merged updating the CI pipelines. Thanks!

**Comment by taylor-swanson** [comment]
I think I'm going to split this PR into smaller chunks. At the very least by codeowner and further by groups of 20 or so packages (like ECS updates).

File: packages/amazon_security_lake/data_stream/event/fields/fields.yml [hunk 1]
```diff
@@ -4492,7 +4492,7 @@
       type: group
       fields:
         - name: classification_ids
-          type: keyword
+          type: integer
```
Comment (efd6) [inline packages/amazon_security_lake/data_stream/event/fields/fields.yml:4495]: Why has this been changed? IDs are usually considered non-orderable.

**Review by efd6 — COMMENTED** [review]
Should we also be considering ssl configurations since they may include keys. Also tenant IDs (leaning towards paranoia).

File: packages/amazon_security_lake/data_stream/event/fields/fields.yml (hunk 1 above)
Comment (taylor-swanson) [reply packages/amazon_security_lake/data_stream/event/fields/fields.yml]: Yeah this was the wrong move. I didn't see that there was a processor that was supposed to convert the this field, but it's not getting triggered on the pipeline test case that is failing:

```
FAILURE DETAILS:
amazon_security_lake/event test-system-activity.log:
[0] parsing field value failed: [0] parsing field value failed: field "ocsf.malware.classification_ids" value "[17 2]" ([]interface {}): expected string or array of strings


╭──────────────────────┬─────────────┬───────────┬───────────────────────────────┬─────────────────────────────────────────────────────────────────────────────┬──────────────╮
│ PACKAGE              │ DATA STREAM │ TEST TYPE │ TEST NAME                     │ RESULT                                                                      │ TIME ELAPSED │
├──────────────────────┼─────────────┼───────────┼───────────────────────────────┼─────────────────────────────────────────────────────────────────────────────┼──────────────┤
│ amazon_security_lake │ event       │ pipeline  │ test-application-activity.log │ PASS                                                                        │  44.295852ms │
│ amazon_security_lake │ event       │ pipeline  │ test-discovery.log            │ PASS                                                                        │   7.062222ms │
│ amazon_security_lake │ event       │ pipeline  │ test-findings.log             │ PASS                                                                        │   9.071077ms │
│ amazon_security_lake │ event       │ pipeline  │ test-iam.log                  │ PASS                                                                        │  16.257727ms │
│ amazon_security_lake │ event       │ pipeline  │ test-network-activity.log     │ PASS                                                                        │  57.253258ms │
│ amazon_security_lake │ event       │ pipeline  │ test-system-activity.log      │ FAIL: test case failed: one or more problems with fields found in documents │  48.515694ms │
╰──────────────────────┴─────────────┴───────────┴───────────────────────────────┴─────────────────────────────────────────────────────────────────────────────┴──────────────╯
```

Looking at the event that is failing, it has (perhaps among other things) a class_id that doesn't match the required list in the default pipeline, so the malware sub-pipeline isn't being executed. I don't have enough context here to know how to fix this problem, so I've downgraded pkg-spec to `3.0.0` so the validation doesn't kick in. It'll have to be addressed in a separate issue.

**Review by efd6 — APPROVED** [review]
Apparently SSL keys handling will be a target of future work.

File: packages/akamai/data_stream/siem/fields/fields.yml [hunk 2]
```diff
@@ -20,6 +20,28 @@
       description: >
         Rules triggered by this request
 
+      fields:
```
Comment (pkoutsovasilis) [inline packages/akamai/data_stream/siem/fields/fields.yml:23]: most probably I miss something here, but I was expecting this PR to only mark sensitive values as secret but this change just adds new fields. Just wondering if we want this to come under this PR or a separate one is wiser

**Comment by taylor-swanson** [comment]
> Apparently SSL keys handling will be a target of future work.

Sorry forgot to reply on that part. Yes, that'll be handled separately.

**Pull request merged**